test/ConfigManagementHelper.js

```diff
@@ -0,0 +1,715 @@
+
+      const expPermissions = expectedPermissionsPerColl[actualColl.name];
+      expect(actualColl.permissions).to.deep.equal(expPermissions);
+
+      if (expColl.type === 'internal') {
+        expect(actualColl.dataLink).to.be.null;
+      } else {
+        expect(actualColl.dataLink).to.exist;
+        expect(actualColl.dataLink.serviceObjectName).to.equal(expColl.serviceObject);
+      }
+    }
+
+    done();
+  });
+};
+
+env.assertCollHooksPerColl = function assertCollHooksPerColl(envId, collName, collHooks, done) {
+  const collHooksNames = Object.keys(collHooks);
+
+  async.eachSeries(
+    collHooksNames,
+    (currentHook, next) => {
+      const backendHookName = CollectionHook[currentHook];
+      ApiService.businessLogic.collHooks.get(envId, collName, backendHookName, (err, actualHook) => {
+        if (err) {
+          return next(err);
+        }
+
+        const expectedHook = collHooks[currentHook];
+        if (expectedHook.type === 'internal') {
+          expect(actualHook.host).to.be.null;
+        } else {
+          expect(actualHook.host).to.not.be.null;
+        }
+
+        const defaultCode = `function ${currentHook}(request, response, modules) {\n  response.continue();\n}`;
+        const expectedCode = expectedHook.code || defaultCode;
+        expect(expectedCode).to.equal(actualHook.code);
+
+        if (expectedHook.type === 'internal') {
+          expect(actualHook.host).to.be.null;
+        } else {
+          expect(actualHook.host).to.not.be.null;
+          expect(actualHook.sdkHandlerName).to.equal(expectedHook.handlerName);
+        }
+
+        next();
+      });
+    },
+    done
+  );
+};
+
+env.assertAllCollHooks = function assertAllCollHooks(envId, configHooks, done) {
+  const collNames = Object.keys(configHooks);
+
+  async.eachSeries(
+    collNames,
+    (currentCollName, next) => {
+      env.assertCollHooksPerColl(envId, currentCollName, configHooks[currentCollName], next);
+    },
+    done
+  );
+};
+
+env.assertEndpoints = function assertEndpoints(envId, configEndpoints, done) {
+  const endpointNames = Object.keys(configEndpoints);
+
+  async.eachSeries(
+    endpointNames,
+    (currentEndpointName, next) => {
+      ApiService.businessLogic.endpoints.get(envId, currentEndpointName, (err, actual) => {
+        if (err) {
+          return done(err);
+        }
+
+        const expected = configEndpoints[currentEndpointName];
+        if (expected.type === 'internal') {
+          expect(actual.host).to.be.null;
+        } else {
+          expect(actual.host).to.not.be.null;
+        }
+
+        const defaultCode = 'function onRequest(request, response, modules) {\n  response.continue();\n}';
+        const expectedCode = expected.code || defaultCode;
+        expect(expectedCode).to.equal(actual.code);
+
+        if (expected.schedule) { // due to api peculiarities
+          if (!expected.schedule.interval) {
+            expect(expected.schedule.start).to.equal(actual.schedule);
```

Comment:
'actual' and 'expected' places should be exchanged. The same is valid for the assertions in the _else_ clause.